src/client/renderer/r_types.h

```diff
@@ -462,7 +462,11 @@ typedef struct {
 	r_bsp_flare_t *flare;
 
 	r_pixel_t lightmap_s, lightmap_t; // lightmap texture coords
-	r_image_t *lightmap;
+
+	union {
+		r_image_t *lightmap;
+		const byte *lightmap_input; // this is only used by r_lightmap
+	};
```

Comment:
Oh, I see (I think). This is the pointer to the sample offset in the lightmap lump? I would either go ahead and waste the pointer on it (because we're going to require the original lightmap samples for stainmaps -- potentially), or add a helper function that returns this pointer for a given surface. Make sense?